### PR TITLE
events, showTitle, markPoint|Line|Area

### LINKED ISSUE #29: markLine removed?
I'm porting a version from the old API to the new one. It's already much nicer! Are you planning to add markLine to the new style DSL at all?

### LINKED ISSUE #42: Add support for `showTitle` in the toolbox
Hi folks,

When hovering over an item in the “toolbox”, Chinese text displays beneath the icon. I’d like to hide this text.

In regular ECharts, you can do this with the `showTitle` field, as shown below. Applying this to the toolbox still allows use of a given feature, and on hover the icon still turns blue, but there is no text displayed below it. This is the behavior I'm looking for.

```
   toolbox: {
      show: true,
      feature: {
         saveAsImage: { show: true }
      },
      showTitle: false
```

You can also see it in the documentation here:
https://ecomfe.github.io/echarts/doc/doc-en.html#Toolbox

I spent a while searching through the docs and source code for the bindings here and could not find the equivalent function.

Have I missed anything? Is there a way to replicate this behavior using these bindings? I’m willing to help, if I can, though I am still very new to PureScript.

### PR DESCRIPTION
**Pull request by cryogenian**
Fixes #42, Fixes #29 

Also added events. 

@safareli Please take a look

### CODE REVIEW

File: src/ECharts/Event.purs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  → Eff (echarts ∷ ECHARTS|e) Unit
+
+
+listenAll
+  ∷ ∀ e
+  . Chart
+  → ( EChartsEvent → Eff (echarts ∷ ECHARTS|e) Unit )
+  → Eff (echarts ∷ ECHARTS|e) Unit
+listenAll chart cb =
+  for_ eventNames \en → on_ chart en \frn →
```
Comment (safareli) [inline src/ECharts/Event.purs:31]: What `frn` stands for?

File: src/ECharts/Event.purs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    for_ (runExcept $ readProp "type" frn >>= readString) \tp →
+      when (tp == en) $ cb $ toEChartsEvent $ Tuple tp frn
+  where
+  eventNames ∷ ∀ rl. RowToList EChartsEventR rl ⇒ VariantTags rl ⇒ L.List String
+  eventNames = variantTags (RLProxy ∷ RLProxy rl)
+
+  toEChartsEvent ∷ ∀ ω. Tuple String ω → EChartsEvent
+  toEChartsEvent = unsafeCoerce
```
Comment (safareli) [inline src/ECharts/Event.purs:39]: Can we do it without unsafe staff?

**Review by safareli — APPROVED** [review]

File: src/ECharts/Event.purs (hunk 1 above)
Comment (cryogenian) [reply src/ECharts/Event.purs:31]: Foreign

File: src/ECharts/Event.purs (hunk 2 above)
Comment (cryogenian) [reply src/ECharts/Event.purs:39]: AFAIK, it's impossible

**Comment by saylu** [comment]
Thanks for this! I've already implemented showTitle in my app.

I do have a brief question -- how do you actually _use_ markLine?

I've tried these:

```haskell
        E.markLine ( E.items $ map (\_ -> ET.numItem 10.0) xs)
        E.markLine $ E.coord 10.0 10.0
        E.markLine $ E.yAxis ...
```

but while some compile, I'm not quite understanding how to use this. In ECharts, you'd provide coordinates or a single number on the yAxis or xAxis or a list of values. Any quick guidance?

File: example/src/Scatter.purs [hunk 3]
```diff
@@ -81,6 +81,15 @@ options sinData cosData = do
 
   E.series do
     E.scatter do
+      E.markPoint do
```
Comment (thomashoneyman) [inline example/src/Scatter.purs:84]: @cryogenian 

I'm able to create a marked point by mimicking this code. I haven't been able to figure out how to use `markLine`, however.

Any guidance on say creating a horizontal line at the number 10.0, for example?

Comment (thomashoneyman) [reply example/src/Scatter.purs:84]: Primarily, I receive some variation on this error:

```
 TypeError: Cannot read property 'coord' of undefined
    at markLineFilter (bundle.js:29146)
    at ...
```

**Pull request merged**